File: functional-tests/base-test.py
import os
from selenium import webdriver
from selenium.webdriver.common.keys import Keys
from selenium.webdriver.common.desired_capabilities import DesiredCapabilities

# Setting up saucelabs config
username = os.environ["SAUCE_USERNAME"]
access_key = os.environ["SAUCE_ACCESS_KEY"]
capabilities = {
    'platform': "Mac OS X 10.9",
    'browserName': "chrome",
    'version': "31",
}
capabilities["tunnel-identifier"] = os.environ["TRAVIS_JOB_NUMBER"]
hub_url = "%s:%s@localhost:4445" % (username, access_key)
driver = webdriver.Remote(desired_capabilities=capabilities, command_executor="https://%s/wd/hub" % hub_url)


# Tests will be run from here
driver.implicitly_wait(5)
driver.get("http://localhost:8000/functional-tests/index.html")
driver.implicitly_wait(10)
elem = driver.find_element_by_class_name("bp-default-logo")
if (not elem):
    raise Exception("Header not found!")

print driver.title

driver.quit()

File: functional-tests/base-test.js
const webdriver = require('selenium-webdriver');
const expect = require('chai').expect;

describe('Base Test', () => {
    before(() => {
        if (process.env.SAUCE_USERNAME && process.env.SAUCE_ACCESS_KEY) {
            this.browser = new webdriver.Builder()
                .usingServer(`http://${process.env.SAUCE_USERNAME}:${process.env.SAUCE_ACCESS_KEY}@ondemand.saucelabs.com:80/wd/hub`)
                .withCapabilities({
                    'tunnel-identifier': process.env.TRAVIS_JOB_NUMBER,
                    build: process.env.TRAVIS_BUILD_NUMBER,
                    username: process.env.SAUCE_USERNAME,
                    accessKey: process.env.SAUCE_ACCESS_KEY,
                    browserName: 'chrome',
                    chromeOptions: {
                        args: ['--disable-web-security']
                    }
                }).build();
        } else {
            this.browser = new webdriver.Builder()
                .withCapabilities({
                    browserName: 'chrome',
                    chromeOptions: {
                        args: ['--disable-web-security']
                    }
                }).build();
        }
    });

    beforeEach(() => {
        return this.browser.get('http://localhost:8000/functional-tests/index.html');
    });

    after(() => {
        return this.browser.quit();
    });

    it('should load a file', () => {
        return this.browser.wait(webdriver.until.elementLocated(webdriver.By.className('bp-loaded')), 5000).then((element) => {
            this.browser.wait(webdriver.until.elementIsVisible(element), 5000).then((el) => {
                expect(el).to.not.equal(undefined);
            });
        });
    });
});
